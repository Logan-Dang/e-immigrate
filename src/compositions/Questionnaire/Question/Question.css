.Question {
    border: none;
    padding: 1rem;
}

.Question .QuestionText {
    font-size: 1rem;
    font-weight: 600;
    margin-bottom: 0.5rem;
}

.QuestionInput:invalid {
    border: 2px solid var(--color-secondary);
}

.Question .RequiredError {
    visibility: hidden;
    color: var(--color-secondary);
}

.TextInput:invalid[data-touched='true'] ~ .RequiredError {
    visibility: visible;
}

.RadioButton:invalid[data-touched='true'] .RequiredError {
    color: blue;
    visibility: visible;
}

.RadioGroup:last-child {
    color: blue;
}
