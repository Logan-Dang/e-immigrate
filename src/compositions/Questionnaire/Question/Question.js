import React, { useState } from 'react';
import { ReactComponent as Arrow } from '../../../data/images/Arrow-Down-Right.svg';

import './Question.css';

import Checkbox from '../../../components/Checkbox/Checkbox';
import Date from '../../../components/Date/Date';
import Email from '../../../components/Email/Email';
import PhoneNumber from '../../../components/PhoneNumber/PhoneNumber';
import Radio from '../../../components/Radio/Radio';
import RadioWithFollowUp from '../../../components/RadioWithFollowUp/RadioWithFollowUp';
import Select from '../../../components/Select/Select';
import TextArea from '../../../components/TextArea/TextArea';
import TextInput from '../../../components/TextInput/TextInput';
import Zip from '../../../components/Zip/Zip';

const Question = ({
    question,
    bindField,
    followUpQuestions,
    collectAnswer,
    setErrors,
    content,
    showFollowUp,
    setShowFollowUp,
}) => {
    const formElements = {
        checkbox: Checkbox,
        date: Date,
        email: Email,
        phoneNumber: PhoneNumber,
        radio: Radio,
        radioWithFollowUp: RadioWithFollowUp,
        dropDown: Select,
        textArea: TextArea,
        input: TextInput,
        zip: Zip,
    };

    const attributes = (q) => {
        return {
            q: q,
            bindField: bindField,
            collectAnswer: collectAnswer,
            content: content,
            answers: q.answerSelections ? q.answerSelections.split(', ') : null,
            selectAnswers: q.answerSelections
                ? ['--', ...q.answerSelections.split(', ')]
                : null,
            values: q.answerSelections ? q.answerValues.split(', ') : null,
            showFollowUp: showFollowUp,
            setShowFollowUp: setShowFollowUp,
            setErrors: setErrors,
        };
    };

    const FormElement = formElements[question.questionType];

    return (
        <>
            <fieldset className="Question">
                <div className="QuestionText">
                    {question.text}
                    {question.required
                        ? ` (${content.required})`
                        : ` (${content.optional})`}
                </div>
                <FormElement attributes={attributes(question)} />
            </fieldset>
            {showFollowUp[question.slug] &&
                followUpQuestions.map((followUpQuestion) => {
                    const FollowUpFormElement =
                        formElements[followUpQuestion.questionType];
                    return (
                        <div className="FollowUp" key={followUpQuestion.slug}>
                            <Arrow height="36px" width="36px" />
                            <fieldset className="Question">
                                <div className="QuestionText">
                                    {followUpQuestion.text}
                                    {followUpQuestion.required
                                        ? ` (${content.required})`
                                        : ` (${content.optional})`}
                                </div>
                                <FollowUpFormElement
                                    attributes={attributes(followUpQuestion)}
                                />
                            </fieldset>
                        </div>
                    );
                })}
        </>
    );
};

export default Question;
