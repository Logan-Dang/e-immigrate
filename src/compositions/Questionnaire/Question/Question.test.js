import React from 'react';
import Question from './Question';

import renderer from 'react-test-renderer';

it('App renders correctly', () => {
    const tree = renderer.create(<Question />).toJSON();
    expect(tree).toMatchSnapshot();
});
