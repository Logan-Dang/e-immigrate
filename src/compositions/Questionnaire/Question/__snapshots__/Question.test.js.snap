// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`question renders correctly 1`] = `
<fieldset
  className="Question"
>
  <div
    className="QuestionText"
  >
    
     (required)
  </div>
  <div>
    An error has occurred. Please return to the home page.
  </div>
</fieldset>
`;
